server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -649,6 +649,14 @@ public interface BlockLoaderContext {
          */
         String indexName();
 
+        /**
+         * Whether the data will be used in stats.
+         * This is relevant to some types, where the choice between doc-values or reading from source is dependent on the usage.
+         * For example, spatial types use doc-values for stats, but read from source for search because doc-values are modified
+         * from the original.
+         */
+        boolean forStats();
```

Comment:
What is confusing for me here is that we'll set this value to false by default, and only set it to true for spatial aggregations loaded from fields. That makes complete sense for the spatial field mappers block loading, but all other types will get a BlockLoaderContext with `preferDocValues==false`, while simultaneously ignoring that, and preferring to load doc-values.

The term `forStats` is not making any statement about whether that implies anything about doc-values versus source loading, and the decision to load differently is entirely local to the field mapper.

All other types can say 'you don't want to use this for stats, well both doc-values and source are equally good enough for non-stats, so I'll load the one I think is fastest'. But if we call it `preferDocValues==false`, then all the other non-spatial types are basically saying "I know I've been told *NOT* to prefer doc-values, but I'm going to ignore that because I magically know this is not relevant to me".

Reply:
How about `useDocValues()`, `loadDocValues`, etc...? My point is `forStats` doesn't tell much.